pennylane_pq/devices.py

```diff
@@ -473,32 +476,38 @@ def pre_expval(self):
         pq.ops.All(pq.ops.Measure) | self._reg #pylint: disable=expression-not-assigned
         self._eng.flush()
 
-    def expval(self, expectation, wires, par):
-        """Retrieve the requested expectation value.
+    def expval(self, observable, wires, par):
+        """Retrieve the requested observable expectation value.
         """
         probabilities = self._eng.backend.get_probabilities(self._reg)
 
-        if expectation == 'PauliX' or expectation == 'PauliY' or expectation == 'PauliZ' or expectation == 'Hadamard':
+        if observable == 'PauliX' or observable == 'PauliY' or observable == 'PauliZ' or observable == 'Hadamard':
 
-            if expectation != 'PauliZ' and not hasattr(self, 'expval_queue'):
-                raise DeviceError("Measurements in basis other than PauliZ are only supported when this plugin is used with versions of PennyLane that expose the expval_queue. Please update PennyLane and this plugin.")
+            if observable != 'PauliZ' and not hasattr(self, 'obs_queue'):
+                raise DeviceError("Measurements in basis other than PauliZ are only supported when this plugin is used with versions of PennyLane that expose the obs_queue. Please update PennyLane and this plugin.")
 
             expval = (1-(2*sum(p for (state, p) in probabilities.items() if state[wires[0]] == '1'))-(1-2*sum(p for (state, p) in probabilities.items() if state[wires[0]] == '0')))/2
-            #variance = 1 - ev**2
-        elif expectation == 'Hermitian':
+
+        elif observable == 'Hermitian':
             raise NotImplementedError
-        elif expectation == 'Identity':
+        elif observable == 'Identity':
             expval = sum(p for (state, p) in probabilities.items())
-            # variance = 1 - expval**2
-        # elif expectation == 'AllPauliZ':
+        # elif observable == 'AllPauliZ':
         #     expval = [((1-2*sum(p for (state, p) in probabilities.items()
         #                         if state[i] == '1'))
         #                -(1-2*sum(p for (state, p) in probabilities.items()
         #                          if state[i] == '0')))/2 for i in range(len(self._reg))]
-        #     variance = [1 - e**2 for e in expval]
 
         return expval
 
+    def var(self, observable, wires, par):
```

Comment:
does this mean we need to pass `par` for every observable, even though none of them support parameters?

Reply:
Yep. This is part of the device level API; an observable with no parameters simply passes `par=[]`. I feel that since this is a lower level API, it is cleaner to be explicit with all parameters.